src/shipit_code_coverage/shipit_code_coverage/report_generators.py

```diff
@@ -39,19 +39,18 @@ def zero_coverage(artifacts, out_dir='code-coverage-reports'):
         if all_lines_uncovered or (len(sf['functions']) > 1 and all_functions_uncovered):
             zero_coverage_files.append(name)
 
-    with open(os.path.join(out_dir, 'zero_coverage_files.json'), 'w') as f:
-        json.dump(zero_coverage_files, f)
-
     mkdir(os.path.join(out_dir, 'zero_coverage_functions'))
 
     zero_coverage_function_counts = []
     for fname, functions in zero_coverage_functions.items():
         zero_coverage_function_counts.append({
             'name': fname,
             'funcs': len(functions),
+            'uncovered': True if fname in zero_coverage_files else False
```

Comment:
The `in` operation for lists is quite expensive. `zero_coverage_files` is a big array and `zero_coverage_functions` contains many elements, so this is going to be quite slow.
The simplest fix is to make `zero_coverage_files` a `set`, in which case `in` is a very fast operation.